src/pages/ProjectSettings.tsx

```diff
@@ -118,18 +118,27 @@ export const ProjectSettings: React.FC = () => {
     ? projectType === ProjectType.Project || repoData
     : repoData;
 
-  const repoDisplay = `${repoData?.repoSettings?.projectRef?.owner}/${repoData?.repoSettings?.projectRef?.repo}`;
+  const { repoSettings } = repoData || {};
+  const { projectRef } = repoSettings || {};
+  const { owner, repo } = projectRef || {};
+
+  const repoDisplay = owner && repo ? `${owner}/${repo}` : identifier;
```

Comment:
Since this variable is passed to `selectedProjectIdentifier`, it is used to determine the display name for a repo or project, not just a repo. And if we're viewing a project, I think it'd be preferable to show the `identifier` instead of owner/repo, no? Currently when looking at http://localhost:3000/project/evergreen/settings/general it shows owner/repo when "evergreen" is what I'd expect to see there.